lib/services/human_time_service.dart

```diff
@@ -0,0 +1,58 @@
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:intl/intl.dart";
+import "package:timeago/timeago.dart" as timeago;
+
+/// A function that returns the current date/time as [DateTime]
+typedef CurrentDateTimeCallback = DateTime Function();
+
+/// Human Time Service is a service that uses dependency injection
+/// to provide related time as a human readable text given a [DateTime]
+class HumanTimeService {
+  static final _absoluteTimeFormatter = DateFormat("EEEE, MMMM d, yyyy HH:mm");
+  static const _relativeLocale = "en_short";
+  static const _agoSuffix = "ago";
+  static const _agoExceptions = ["now"];
+
+  final CurrentDateTimeCallback currentDateTimeCallback;
+
+  HumanTimeService({
+    required this.currentDateTimeCallback,
+  });
+
+  /// Converts a [DateTime] to a human readable time ago string
+  /// Examples of format: "5m ago", "now", "2 days ago"
+  String textTimeSince(DateTime dateTime) {
+    final referenceClock = currentDateTimeCallback();
+
+    final textTime = timeago.format(
+      dateTime,
+      locale: _relativeLocale,
+      clock: referenceClock,
+    );
+
+    if (_agoExceptions.contains(textTime)) {
```

Comment:
Here the contains method only checks one corner case ("now"). Could we use a String "_agoException"  instead of a List<String> "_agoExceptions". In this case we would just check if "_agoException" is equal to "textTime".

If other exceptions will be tested in the future, then I understand the current implementation.

Reply:
Yes maybe we will discover other special cases (like "now") in the future. This is why I implemented it like this.

Reply:
Ok so we keep it like that!